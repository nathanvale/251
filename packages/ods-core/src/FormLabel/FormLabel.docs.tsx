/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { FormLabel, FormLabelProps } from "./FormLabel";

export const docs: ComponentDocs<FormLabelProps> = {
  category: "Atomic",
  componentName: "FormLabel",
  description:
    "This is a low level and atomic component used to provide a label for interactive form components.",
  propDescriptions: {
    id: "The unique id of the component, required to accessibility",
    children: ["The text to show for the label."].join(" "),
    component: ["The html element type to be used under the hood."].join(" "),
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <FormLabel>This is a label</FormLabel>,
    },
    additional: [
      {
        label: "Custom component",
        description: ["Label"].join(" "),
        Code: () => <FormLabel component="legend">This is a legend</FormLabel>,
      },
    ],
  },
  snippets: [
    {
      label: "Simple",
      Code: () => <FormLabel>Some label</FormLabel>,
    },
  ],
};
