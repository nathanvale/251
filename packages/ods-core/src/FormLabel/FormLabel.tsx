import React from "react";
import clsx from "clsx";
import {
  FormLabel as MuiFormLabel,
  FormLabelProps as MuiFormLabelProps,
  makeStyles,
} from "@material-ui/core";
import { BaseFormStateProps, LabelElements } from "@origin-digital/ods-types";

export interface FormLabelProps extends BaseFormStateProps {
  component?: LabelElements;
  muiProps?: MuiFormLabelProps;
}

/**
 * In MUI FormLabel gets the palette.primary.main colour when it is focused on, however we do not want our
 * labels to change colour when they are focused.
 */
const useLabelStyles = makeStyles(
  (theme) => ({
    root: {
      "&.Mui-error": {
        color: theme.palette.text.primary,
      },
      "&$focused": {
        color: theme.palette.text.primary,
      },
      // In Material-UI error has priority over disabled. However, for us disabled should override others as the highest priority.
      "&.Mui-disabled": {
        color: theme.palette.text.disabled,
      },
    },
    focused: {},
  }),
  { classNamePrefix: "FormLabel" }
);

export const FormLabel = ({
  disabled,
  focused,
  muiProps = {} as any,
  classes,
  className,
  ...others
}: FormLabelProps) => {
  const lblClasses = useLabelStyles();
  return (
    <MuiFormLabel
      {...muiProps}
      {...others}
      className={clsx(
        lblClasses.root,
        {
          [lblClasses.focused]: focused,
        },
        className
      )}
      classes={classes}
      disabled={disabled}
      focused={focused}
    />
  );
};

FormLabel.displayName = "FormLabel";
FormLabel.defaultProps = {
  component: "label",
  "data-id": "formLabel",
};
