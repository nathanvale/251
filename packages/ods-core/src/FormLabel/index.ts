export { FormLabel } from "./FormLabel";
export type FormLabelProps = import("./FormLabel").FormLabelProps;
