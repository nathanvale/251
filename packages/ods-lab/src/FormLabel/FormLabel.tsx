import * as React from "react";
import MuiFormLabel, {
  FormLabelProps as IFormLabel,
} from "@material-ui/core/FormLabel";

export interface FormLabelProps extends IFormLabel {
  "data-id"?: string;
  component?: React.ElementType;
}

export const FormLabel = (props: FormLabelProps) => (
  <MuiFormLabel {...props}>{props.children}</MuiFormLabel>
);
