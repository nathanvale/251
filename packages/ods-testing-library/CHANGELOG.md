# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
