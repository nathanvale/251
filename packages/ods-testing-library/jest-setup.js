import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
