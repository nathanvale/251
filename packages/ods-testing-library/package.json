{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
