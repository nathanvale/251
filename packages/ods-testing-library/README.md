# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
