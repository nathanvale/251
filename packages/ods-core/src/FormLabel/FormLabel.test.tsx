import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { FormLabel } from "./FormLabel";

test("renders a label with default data-id", () => {
  const { container } = render(<FormLabel id="mylbl">Some label</FormLabel>);

  expect(container).toMatchSnapshot();
});

test("renders a legend element", () => {
  const { container } = render(
    <FormLabel id="mylbl" component="legend">
      Some label
    </FormLabel>,
  );

  expect(container.getElementsByTagName("label")).toHaveLength(0);
  expect(container.getElementsByTagName("legend")).toHaveLength(1);
});

test("When the label is focused, the color remains grey.", () => {
  const { container } = render(<FormLabel id="mylbl">Some label</FormLabel>);
  const labelEl = container.getElementsByTagName("label")[0];
  let styles = window.getComputedStyle(labelEl);
  expect(styles.color).toBe("rgb(80, 80, 80)");

  labelEl.focus();

  styles = window.getComputedStyle(labelEl);
  expect(styles.color).toBe("rgb(80, 80, 80)");
});
