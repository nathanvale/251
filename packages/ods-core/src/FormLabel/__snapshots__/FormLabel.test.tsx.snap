// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a label with default data-id 1`] = `
<div>
  <label
    class="MuiFormLabel-root FormLabel-root-1"
    data-id="formLabel"
    id="mylbl"
  >
    Some label
  </label>
</div>
`;
